### ISSUE TITLE
`comparisons` does not display all comparisons of `fixest` model

### ISSUE THREAD

**Issue by maellecoursonnais** [vincentarelbundock/marginaleffects#493]
Hello again, 

I am trying to compute comparisons with a `fixest` model, but it seems that `comparisons` only displays contrasts between the first and second values and not the others with a `fixest` model. 

MWE: 
```r
library(marginaleffects)
library(fixest)
mod <- feglm(vs ~ hp * factor(cyl), data = mtcars, family = "binomial")

pred <- comparisons(model = mod, 
                    variables = "cyl",
                    by = "hp",
                    newdata = datagrid(cyl = unique,
                                       hp = seq(80, 110, 10)))
```
gives
```r
#> pred
#       type term          contrast  comparison std.error  statistic   p.value   conf.low conf.high predicted predicted_hi predicted_lo  hp
# 1 response  cyl mean(6) - mean(4) -0.05892818 0.2681428 -0.2197641 0.8260548 -0.5844785 0.4666221 0.8623183    0.8623183    0.9212465  80
# 2 response  cyl mean(6) - mean(4) -0.08992845 0.2768887 -0.3247820 0.7453461 -0.6326203 0.4527634 0.8119314    0.8119314    0.9018599  90
# 3 response  cyl mean(6) - mean(4) -0.12984808 0.2811578 -0.4618334 0.6442008 -0.6809073 0.4212111 0.7484831    0.7484831    0.8783312 100
# 4 response  cyl mean(6) - mean(4) -0.17782968 0.3061520 -0.5808542 0.5613387 -0.7778766 0.4222172 0.6722695    0.6722695    0.8500991 110
```
, while the `glm` equivalent gives:
```r
> mod <- glm(vs ~ hp * factor(cyl), data = mtcars, family = "binomial")
> comparisons(model = mod, variables = "cyl", by = "hp",
+             newdata = datagrid(cyl = unique,
+                                        hp = seq(80, 110, 10)))
#       type term          contrast  comparison  std.error   statistic      p.value   conf.low  conf.high predicted predicted_hi predicted_lo  hp
# 1 response  cyl mean(6) - mean(4) -0.05892818 0.24556815  -0.2399667 8.103561e-01 -0.5402329  0.4223765 0.8623183 8.623183e-01    0.9212465  80
# 2 response  cyl mean(6) - mean(4) -0.08992845 0.25357768  -0.3546387 7.228603e-01 -0.5869316  0.4070747 0.8119314 8.119314e-01    0.9018599  90
# 3 response  cyl mean(6) - mean(4) -0.12984808 0.25748742  -0.5042890 6.140583e-01 -0.6345142  0.3748180 0.7484831 7.484831e-01    0.8783312 100
# 4 response  cyl mean(6) - mean(4) -0.17782968 0.28037736  -0.6342512 5.259169e-01 -0.7273592  0.3716999 0.6722695 6.722695e-01    0.8500991 110
# 5 response  cyl mean(8) - mean(4) -0.92124650 0.08787501 -10.4836006 1.027555e-25 -1.0934784 -0.7490146 0.8623183 1.170226e-09    0.9212465  80
# 6 response  cyl mean(8) - mean(4) -0.90185988 0.09370755  -9.6241968 6.319874e-22 -1.0855233 -0.7181965 0.8119314 1.170226e-09    0.9018599  90
# 7 response  cyl mean(8) - mean(4) -0.87833117 0.12643264  -6.9470289 3.730594e-12 -1.1261346 -0.6305278 0.7484831 1.170226e-09    0.8783312 100
# 8 response  cyl mean(8) - mean(4) -0.85009914 0.19344816  -4.3944544 1.110513e-05 -1.2292506 -0.4709477 0.6722695 1.170226e-09    0.8500991 110
```

**Comment by vincentarelbundock** [comment]
Thanks a lot for the report, I really appreciate it! This bug should be fixed by: 976f8d583f87e99c7d08ab06a748c82a9196e39e

**Comment by maellecoursonnais** [comment]
Works like a charm now! Thanks
